helix-term/src/commands.rs

```diff
@@ -1787,26 +1787,58 @@ mod cmd {
         Ok(())
     }
 
+    fn buffer_close_impl(editor: &mut Editor, args: &[&str], force: bool) -> anyhow::Result<()> {
+        if args.is_empty() {
+            let doc_id = view!(editor).doc;
+            editor.close_document(doc_id, force)?;
+            return Ok(());
+        }
+
+        let mut nonexistent_buffers = vec![];
+        for arg in args {
+            let doc_id = editor.documents().find_map(|doc| {
+                let arg_path = Some(Path::new(arg));
+                if doc.path().map(|p| p.as_path()) == arg_path
+                    || doc.relative_path().as_deref() == arg_path
+                {
+                    Some(doc.id())
+                } else {
+                    None
+                }
+            });
+
+            match doc_id {
+                Some(doc_id) => editor.close_document(doc_id, force)?,
+                None => nonexistent_buffers.push(arg),
+            }
+        }
+
+        let nonexistent_buffers: Vec<_> = nonexistent_buffers
+            .into_iter()
+            .map(|str| format!("'{}'", str))
+            .collect();
+        editor.set_error(format!(
+            "couldn't close buffer(s) {}: does not exist",
+            nonexistent_buffers.join(", ")
+        ));
+
+        Ok(())
+    }
```

Comment:
![Screenshot_20211211_175512](https://user-images.githubusercontent.com/4687791/145672346-d5f15a85-2943-4f62-9363-cbac6c4e3f55.png)

Is it useful to show scratch buffer but can't be deleted?